gfx.cpp

```diff
@@ -6510,12 +6493,10 @@ class GfxInternal
             did_type_str = "uint3";
             break;
         case GfxTexture::kType_3D:
+        default:
```

Comment:
Any reason for moving the `default` here?

It's useful to assert on unsupported switch/case values, like when you add in a new enum entry and you can then follow the asserts for the various switch/case statements that have to be updated accordingly.

Reply:
It was throwing a bunch of errors about uninitialised variables etc. when taking the default path, so I made default also the 3D path. Again this could be fixed by having assert terminate, or use static assert instead. std::unreachable is only available in c++23 but we could create our own macro implementation to keep the compiler happy

Reply:
Fixed now that assert terminates